backend/app/Http/Controllers/Statistic/ShowUserStatisticController.php

```diff
@@ -18,6 +26,106 @@ class ShowUserStatisticController extends Controller
      */
     public function __invoke(): View|Factory
     {
-        return view('statistic.user', []);
+        /**
+         * ユーザーランクごとの情報取得
+         * @var User
+         */
+        $perUserRoles = User::groupBy('user_role_id')->select('user_role_id', DB::raw('count(*) as user_role_count'))->get();
+        /** @var UserRole */
+        $userRoles = UserRole::all();
+        /** @var array<string,int> 権限名:数 の連想配列 */
+        $userRoleCounter = [];
+        foreach ($perUserRoles as $perUserRole) {
+            //
+            $userRoleCounter[$userRoles->where('id', $perUserRole->user_role_id)->first()->name] = $perUserRole->user_role_count;
+        }
+
+        /**
+         * 学年ごと、学部ごと、学科ごと、出身地、現住地、性別ごとの取得
+         * @var UserInfo
+         */
+        $perUserInfos = UserInfo::groupBy('grade', 'u_u_major_id', 'birth_area_id', 'live_area_id', 'gender')
+            ->select(['grade', DB::raw('count(*) as grade_count'), 'u_u_major_id', DB::raw('count(*) as u_u_major_count'), 'birth_area_id', DB::raw('count(*) as birth_area_count'), 'live_area_id', DB::raw('count(*) as live_area_count'), 'gender', DB::raw('count(*) as gender_count')])
+            ->get();
+        /** @var array<string,int> 学年名:数 の連想配列 */
+        $userGradeCounter = [];
+        /** @var array<string,int> 性別名:数 の連想配列 */
+        $userGenderCounter = [];
+        /** @var array<string,int> 学部名:数 の連想配列 */
+        $userFacultyCounter = [];
+        /** @var array<string,int> 学科名:数 の連想配列 */
+        $userMajorCounter = [];
+        /** @var array<string,int> 現住地:数 の連想配列 */
+        $userLiveAreaCounter = [];
+        /** @var array<string,int> 出身地:数 の連想配列 */
+        $userBirthAreaCounter = [];
+
+        /**予め取得 */
+        $uuMajors = UUMajor::all();
+        foreach ($perUserInfos as $perUserInfo) {
+            /**
+             * 学年処理
+             */
+            $userGradeCounter[$perUserInfo->grade->label()] = $perUserInfo->grade_count;
+            /**
+             * 性別処理
+             */
+            $userGenderCounter[$perUserInfo->gender->label()] = $perUserInfo->gender_count;
+
+            /**
+             * 学部学科処理
+             */
+            //学部はenumに変換して加算する
+            if ($perUserInfo->u_u_major_id) {
+                $major = $uuMajors->where('id', $perUserInfo->u_u_major_id)->first();
+                \Log::debug($major);
```

Comment:
**[phpstan]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Parameter #1 $message of static method Illuminate\Log\Logger::debug() expects array|Illuminate\Contracts\Support\Arrayable|Illuminate\Contracts\Support\Jsonable|Illuminate\Support\Stringable|string, App\Models\UUMajor|null given.